awx/main/scheduler/task_manager.py

```diff
@@ -510,13 +487,49 @@ def process_pending_tasks(self, pending_tasks):
                 self.start_task(task, None, task.get_jobs_fail_chain(), None)
                 continue
 
+            # Determine if ther is control capacity for the task
+            remaining_control_capacity = self.get_remaining_capacity('controlplane', capacity_type='control')
+            if remaining_control_capacity < settings.AWX_CONTROL_PLANE_TASK_IMPACT:
+                logger.debug(f"Skipping task {task.log_format} in pending, not enough capacity left on controlplane to control new tasks")
+                continue
```

Comment:
I'd vote to delete this condition. I was making notes myself to delete the similar condition for `execution_node` selection. If the IG doesn't have capacity, then the `fit_task_to_most_remaining_capacity_instance` / `find_largest_idle_instance` should fail to find a node. I know I asked you to back out your large refactor for now, but I want to hold onto the important critical thinking that came from it.

Reply:
ok I've done this